crypto/bn/bn_lib.c

```diff
@@ -446,6 +446,15 @@ static BIGNUM *bin2bn(const unsigned char *s, int len, BIGNUM *ret,
         return NULL;
     bn_check_top(ret);
 
+    /*
+     * If the input has no bits, the number is considered zero.
+     * This makes calls with s==NULL and len==0 safe.
+     */
+    if (len == 0) {
```

Comment:
`len <= 0` ?
Or check separately that `len < 0` and return an error if so.  I'm okay with either approach.